2018-03-09 14:03:38,549 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:03:38,634 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:03:38,800 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:03:38,955 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:03:38,996 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:03:39,002 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:03:39,008 - Server - INFO - Started up on port 50000
2018-03-09 14:03:39,026 - Server - INFO - Thread: 4736 - Start requesting peers. 2 peers currently
2018-03-09 14:03:39,052 - Server - WARNING - Thread: 5072 - Started updating from peers
2018-03-09 14:03:39,185 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:03:39,192 - Server - WARNING - Thread: 4736 - Complete requesting peers. 2 peers currently
2018-03-09 14:03:39,271 - Server - INFO - Thread: 5072 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:03:39,278 - Server - WARNING - Thread: 5072 - Completed updating from peers
2018-03-09 14:04:00,374 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:04:00,550 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:04:00,718 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:04:00,877 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:04:00,915 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:04:00,921 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:04:00,934 - Server - INFO - Started up on port 50000
2018-03-09 14:04:00,951 - Server - INFO - Thread: 5120 - Start requesting peers. 2 peers currently
2018-03-09 14:04:00,978 - Server - WARNING - Thread: 5456 - Started updating from peers
2018-03-09 14:04:01,122 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:04:01,128 - Server - WARNING - Thread: 5120 - Complete requesting peers. 2 peers currently
2018-03-09 14:04:01,242 - Server - INFO - Thread: 5456 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:04:01,248 - Server - WARNING - Thread: 5456 - Completed updating from peers
2018-03-09 14:05:15,468 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:05:15,586 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:05:15,739 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:05:15,893 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:05:15,934 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:05:15,941 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:05:15,965 - Server - INFO - Started up on port 50000
2018-03-09 14:05:15,984 - Server - INFO - Thread: 8272 - Start requesting peers. 2 peers currently
2018-03-09 14:05:16,016 - Server - WARNING - Thread: 8608 - Started updating from peers
2018-03-09 14:05:16,161 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:05:16,167 - Server - WARNING - Thread: 8272 - Complete requesting peers. 2 peers currently
2018-03-09 14:05:16,275 - Server - INFO - Thread: 8608 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:05:16,281 - Server - WARNING - Thread: 8608 - Completed updating from peers
2018-03-09 14:05:23,479 - Server - INFO - Thread: 8272 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:41864 - Received duplicate or invalid message, ignoring
2018-03-09 14:07:21,913 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:07:22,068 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:07:22,222 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:07:22,377 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:07:22,414 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:07:22,420 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:07:22,443 - Server - INFO - Started up on port 50000
2018-03-09 14:07:22,460 - Server - INFO - Thread: 6848 - Start requesting peers. 2 peers currently
2018-03-09 14:07:22,490 - Server - WARNING - Thread: 7184 - Started updating from peers
2018-03-09 14:07:22,620 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:07:22,626 - Server - WARNING - Thread: 6848 - Complete requesting peers. 2 peers currently
2018-03-09 14:07:22,744 - Server - INFO - Thread: 7184 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:07:22,750 - Server - WARNING - Thread: 7184 - Completed updating from peers
2018-03-09 14:21:52,282 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:21:53,785 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:21:53,948 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:21:54,092 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:21:54,124 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:21:54,129 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:21:54,153 - Server - INFO - Started up on port 50000
2018-03-09 14:21:54,171 - Server - INFO - Thread: 6784 - Start requesting peers. 2 peers currently
2018-03-09 14:21:54,188 - Server - WARNING - Thread: 7120 - Started updating from peers
2018-03-09 14:21:54,325 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:21:54,331 - Server - WARNING - Thread: 6784 - Complete requesting peers. 2 peers currently
2018-03-09 14:21:54,434 - Server - INFO - Thread: 7120 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:21:54,439 - Server - WARNING - Thread: 7120 - Completed updating from peers
2018-03-09 14:21:55,784 - Server - INFO - Thread: 6784 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:42334 - Received duplicate or invalid message, ignoring
2018-03-09 14:28:14,884 - Server - WARNING - =========== Server logging started ==========
2018-03-09 14:28:15,118 - Server - INFO - Reading peers from peers.txt.
2018-03-09 14:28:15,310 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 14:28:15,439 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 14:28:15,478 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:28:15,479 - Server - WARNING - ============ Server init complete ===========
2018-03-09 14:28:15,480 - Server - INFO - Started up on port 50000
2018-03-09 14:28:15,481 - Server - INFO - Thread: 6960 - Start requesting peers. 2 peers currently
2018-03-09 14:28:15,493 - Server - WARNING - Thread: 7296 - Started updating from peers
2018-03-09 14:28:15,614 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 14:28:15,615 - Server - WARNING - Thread: 6960 - Complete requesting peers. 2 peers currently
2018-03-09 14:28:15,642 - Server - INFO - Thread: 7296 - Received 11 blocks from 149.106.60.81.
2018-03-09 14:28:15,643 - Server - WARNING - Thread: 7296 - Completed updating from peers
2018-03-09 14:28:25,426 - Server - INFO - Thread: 6960 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:42628 - Received duplicate or invalid message, ignoring
2018-03-09 16:10:36,129 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:10:36,267 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:10:36,449 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:10:36,607 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:10:36,642 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:10:36,647 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:10:36,664 - Server - INFO - Started up on port 50000
2018-03-09 16:10:36,683 - Server - INFO - Thread: 1360 - Start requesting peers. 2 peers currently
2018-03-09 16:10:36,697 - Server - WARNING - Thread: 1696 - Started updating from peers
2018-03-09 16:10:36,831 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:10:36,831 - Server - WARNING - Thread: 1360 - Complete requesting peers. 2 peers currently
2018-03-09 16:10:36,945 - Server - INFO - Thread: 1696 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:10:36,951 - Server - WARNING - Thread: 1696 - Completed updating from peers
2018-03-09 16:10:49,670 - Server - INFO - Thread: 1360 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:46628 - Received duplicate or invalid message, ignoring
2018-03-09 16:13:28,190 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:13:28,332 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:13:28,495 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:13:28,640 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:13:28,677 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:13:28,683 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:13:28,712 - Server - INFO - Started up on port 50000
2018-03-09 16:13:28,732 - Server - INFO - Thread: 928 - Start requesting peers. 2 peers currently
2018-03-09 16:13:28,765 - Server - WARNING - Thread: 1264 - Started updating from peers
2018-03-09 16:13:28,887 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:13:28,893 - Server - WARNING - Thread: 928 - Complete requesting peers. 2 peers currently
2018-03-09 16:13:29,011 - Server - INFO - Thread: 1264 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:13:29,017 - Server - WARNING - Thread: 1264 - Completed updating from peers
2018-03-09 16:13:34,901 - Server - INFO - Thread: 928 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:46762 - Received duplicate or invalid message, ignoring
2018-03-09 16:15:05,129 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:15:05,266 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:15:05,427 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:15:05,569 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:15:05,599 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:15:05,605 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:15:05,628 - Server - INFO - Started up on port 50000
2018-03-09 16:15:05,634 - Server - INFO - Thread: 7584 - Start requesting peers. 2 peers currently
2018-03-09 16:15:05,659 - Server - WARNING - Thread: 7920 - Started updating from peers
2018-03-09 16:15:05,793 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:15:05,799 - Server - WARNING - Thread: 7584 - Complete requesting peers. 2 peers currently
2018-03-09 16:15:05,891 - Server - INFO - Thread: 7920 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:15:05,896 - Server - WARNING - Thread: 7920 - Completed updating from peers
2018-03-09 16:15:20,118 - Server - ERROR - Thread: 7584 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:46846 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 20, in print
    print("PUB_KEY: ",binascii.unhexlify(self.pub_key))
binascii.Error: Non-hexadecimal digit found

2018-03-09 16:37:51,263 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:37:51,380 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:37:51,535 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:37:51,690 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:37:51,728 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:37:51,734 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:37:51,756 - Server - INFO - Started up on port 50000
2018-03-09 16:37:51,774 - Server - INFO - Thread: 6656 - Start requesting peers. 2 peers currently
2018-03-09 16:37:51,799 - Server - WARNING - Thread: 6992 - Started updating from peers
2018-03-09 16:37:51,948 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:37:51,955 - Server - WARNING - Thread: 6656 - Complete requesting peers. 2 peers currently
2018-03-09 16:37:52,052 - Server - INFO - Thread: 6992 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:37:52,059 - Server - WARNING - Thread: 6992 - Completed updating from peers
2018-03-09 16:37:52,402 - Server - INFO - Thread: 6656 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:47914 - Received duplicate or invalid message, ignoring
2018-03-09 16:42:33,997 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:42:34,147 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:42:34,290 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:42:34,434 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:42:34,471 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:42:34,478 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:42:34,501 - Server - INFO - Started up on port 50000
2018-03-09 16:42:34,521 - Server - INFO - Thread: 5840 - Start requesting peers. 2 peers currently
2018-03-09 16:42:34,554 - Server - WARNING - Thread: 6176 - Started updating from peers
2018-03-09 16:42:34,682 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:42:34,689 - Server - WARNING - Thread: 5840 - Complete requesting peers. 2 peers currently
2018-03-09 16:42:34,811 - Server - INFO - Thread: 6176 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:42:34,817 - Server - WARNING - Thread: 6176 - Completed updating from peers
2018-03-09 16:42:37,610 - Server - INFO - Thread: 5840 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48244 - Received duplicate or invalid message, ignoring
2018-03-09 16:42:52,650 - Server - INFO - Thread: 5840 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48262 - Received duplicate or invalid message, ignoring
2018-03-09 16:47:12,953 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:47:13,039 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:47:13,207 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:47:13,356 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:47:13,398 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:47:13,398 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:47:13,410 - Server - INFO - Started up on port 50000
2018-03-09 16:47:13,428 - Server - INFO - Thread: 3968 - Start requesting peers. 2 peers currently
2018-03-09 16:47:13,452 - Server - WARNING - Thread: 4304 - Started updating from peers
2018-03-09 16:47:13,604 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:47:13,610 - Server - WARNING - Thread: 3968 - Complete requesting peers. 2 peers currently
2018-03-09 16:47:13,694 - Server - INFO - Thread: 4304 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:47:13,701 - Server - WARNING - Thread: 4304 - Completed updating from peers
2018-03-09 16:47:23,615 - Server - ERROR - Thread: 3968 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48530 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    self.pub_key.verify(
AttributeError: 'bytes' object has no attribute 'verify'

2018-03-09 16:47:37,561 - Server - ERROR - Thread: 3968 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48544 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    self.pub_key.verify(
AttributeError: 'bytes' object has no attribute 'verify'

2018-03-09 16:47:52,550 - Server - ERROR - Thread: 3968 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48558 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    self.pub_key.verify(
AttributeError: 'bytes' object has no attribute 'verify'

2018-03-09 16:48:20,683 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:48:20,833 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:48:20,985 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:48:21,135 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:48:21,171 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:48:21,178 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:48:21,196 - Server - INFO - Started up on port 50000
2018-03-09 16:48:21,214 - Server - INFO - Thread: 4528 - Start requesting peers. 2 peers currently
2018-03-09 16:48:21,248 - Server - WARNING - Thread: 4864 - Started updating from peers
2018-03-09 16:48:21,376 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:48:21,382 - Server - WARNING - Thread: 4528 - Complete requesting peers. 2 peers currently
2018-03-09 16:48:21,515 - Server - INFO - Thread: 4864 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:48:21,520 - Server - WARNING - Thread: 4864 - Completed updating from peers
2018-03-09 16:48:22,680 - Server - ERROR - Thread: 4528 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:48588 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = self.pub_key.public_key()
AttributeError: 'bytes' object has no attribute 'public_key'

2018-03-09 16:55:51,272 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:55:51,387 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:55:51,539 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:55:51,694 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:55:51,730 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:55:51,735 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:55:51,758 - Server - INFO - Started up on port 50000
2018-03-09 16:55:51,776 - Server - INFO - Thread: 432 - Start requesting peers. 2 peers currently
2018-03-09 16:55:51,794 - Server - WARNING - Thread: 768 - Started updating from peers
2018-03-09 16:55:51,947 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:55:51,953 - Server - WARNING - Thread: 432 - Complete requesting peers. 2 peers currently
2018-03-09 16:55:52,048 - Server - INFO - Thread: 768 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:55:52,054 - Server - WARNING - Thread: 768 - Completed updating from peers
2018-03-09 16:55:54,726 - Server - ERROR - Thread: 432 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49012 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = rsa.importKey(self.pub_key)
AttributeError: 'module' object has no attribute 'importKey'

2018-03-09 16:56:08,808 - Server - ERROR - Thread: 432 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49028 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = rsa.importKey(self.pub_key)
AttributeError: 'module' object has no attribute 'importKey'

2018-03-09 16:59:13,123 - Server - WARNING - =========== Server logging started ==========
2018-03-09 16:59:13,272 - Server - INFO - Reading peers from peers.txt.
2018-03-09 16:59:13,425 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 16:59:13,572 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 16:59:13,613 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:59:13,619 - Server - WARNING - ============ Server init complete ===========
2018-03-09 16:59:13,644 - Server - INFO - Started up on port 50000
2018-03-09 16:59:13,663 - Server - INFO - Thread: 2448 - Start requesting peers. 2 peers currently
2018-03-09 16:59:13,694 - Server - WARNING - Thread: 2784 - Started updating from peers
2018-03-09 16:59:13,814 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 16:59:13,820 - Server - WARNING - Thread: 2448 - Complete requesting peers. 2 peers currently
2018-03-09 16:59:13,948 - Server - INFO - Thread: 2784 - Received 11 blocks from 149.106.60.81.
2018-03-09 16:59:13,954 - Server - WARNING - Thread: 2784 - Completed updating from peers
2018-03-09 16:59:24,154 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49222 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 16:59:39,161 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49238 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 16:59:54,213 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49254 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:00:09,241 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49270 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:00:24,292 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49286 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:00:39,132 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49302 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:00:54,066 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49316 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:01:09,088 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49330 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:01:24,111 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49344 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:01:39,194 - Server - ERROR - Thread: 2448 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49358 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key)
TypeError: load_pem_public_key() missing 1 required positional argument: 'backend'

2018-03-09 17:01:54,232 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:01:54,309 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:01:54,476 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:01:54,629 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:01:54,670 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:01:54,676 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:01:54,700 - Server - INFO - Started up on port 50000
2018-03-09 17:01:54,718 - Server - INFO - Thread: 7264 - Start requesting peers. 2 peers currently
2018-03-09 17:01:54,738 - Server - WARNING - Thread: 7600 - Started updating from peers
2018-03-09 17:01:54,872 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:01:54,878 - Server - WARNING - Thread: 7264 - Complete requesting peers. 2 peers currently
2018-03-09 17:01:54,984 - Server - INFO - Thread: 7600 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:01:54,991 - Server - WARNING - Thread: 7600 - Completed updating from peers
2018-03-09 17:02:12,613 - Server - ERROR - Thread: 7264 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49386 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:02:36,580 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:02:36,685 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:02:36,860 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:02:37,009 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:02:37,045 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:02:37,050 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:02:37,081 - Server - INFO - Started up on port 50000
2018-03-09 17:02:37,099 - Server - INFO - Thread: 4240 - Start requesting peers. 2 peers currently
2018-03-09 17:02:37,120 - Server - WARNING - Thread: 4576 - Started updating from peers
2018-03-09 17:02:37,258 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:02:37,264 - Server - WARNING - Thread: 4240 - Complete requesting peers. 2 peers currently
2018-03-09 17:02:37,355 - Server - INFO - Thread: 4576 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:02:37,362 - Server - WARNING - Thread: 4576 - Completed updating from peers
2018-03-09 17:02:41,540 - Server - ERROR - Thread: 4240 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49414 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:05:00,112 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:05:00,221 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:05:00,384 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:05:00,532 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:05:00,565 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:05:00,571 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:05:00,602 - Server - INFO - Started up on port 50000
2018-03-09 17:05:00,619 - Server - INFO - Thread: 8912 - Start requesting peers. 2 peers currently
2018-03-09 17:05:00,639 - Server - WARNING - Thread: 9248 - Started updating from peers
2018-03-09 17:05:00,779 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:05:00,785 - Server - WARNING - Thread: 8912 - Complete requesting peers. 2 peers currently
2018-03-09 17:05:00,877 - Server - INFO - Thread: 9248 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:05:00,883 - Server - WARNING - Thread: 9248 - Completed updating from peers
2018-03-09 17:05:11,967 - Server - ERROR - Thread: 8912 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49530 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 28, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:06:29,431 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:06:29,546 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:06:29,712 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:06:29,872 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:06:29,910 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:06:29,916 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:06:29,934 - Server - INFO - Started up on port 50000
2018-03-09 17:06:29,952 - Server - INFO - Thread: 7840 - Start requesting peers. 2 peers currently
2018-03-09 17:06:29,976 - Server - WARNING - Thread: 8176 - Started updating from peers
2018-03-09 17:06:30,120 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:06:30,125 - Server - WARNING - Thread: 7840 - Complete requesting peers. 2 peers currently
2018-03-09 17:06:30,225 - Server - INFO - Thread: 8176 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:06:30,232 - Server - WARNING - Thread: 8176 - Completed updating from peers
2018-03-09 17:06:42,260 - Server - ERROR - Thread: 7840 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49596 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 29, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:07:10,453 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:07:10,555 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:07:10,729 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:07:10,871 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:07:10,914 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:07:10,921 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:07:10,933 - Server - INFO - Started up on port 50000
2018-03-09 17:07:10,952 - Server - INFO - Thread: 7440 - Start requesting peers. 2 peers currently
2018-03-09 17:07:10,974 - Server - WARNING - Thread: 7776 - Started updating from peers
2018-03-09 17:07:11,113 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:07:11,120 - Server - WARNING - Thread: 7440 - Complete requesting peers. 2 peers currently
2018-03-09 17:07:11,213 - Server - INFO - Thread: 7776 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:07:11,220 - Server - WARNING - Thread: 7776 - Completed updating from peers
2018-03-09 17:07:27,275 - Server - ERROR - Thread: 7440 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49632 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 29, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:08:16,601 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:08:16,703 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:08:16,856 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:08:17,019 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:08:17,059 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:08:17,065 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:08:17,084 - Server - ERROR - Unable to start on port 50000
2018-03-09 17:08:22,258 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:08:22,389 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:08:22,552 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:08:22,715 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:08:22,752 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:08:22,758 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:08:22,782 - Server - INFO - Started up on port 50000
2018-03-09 17:08:22,800 - Server - INFO - Thread: 2496 - Start requesting peers. 2 peers currently
2018-03-09 17:08:22,825 - Server - WARNING - Thread: 2832 - Started updating from peers
2018-03-09 17:08:22,973 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:08:22,979 - Server - WARNING - Thread: 2496 - Complete requesting peers. 2 peers currently
2018-03-09 17:08:23,053 - Server - INFO - Thread: 2832 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:08:23,059 - Server - WARNING - Thread: 2832 - Completed updating from peers
2018-03-09 17:08:27,208 - Server - ERROR - Thread: 2496 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49680 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 29, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:09:48,732 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:09:48,867 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:09:49,026 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:09:49,173 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:09:49,217 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:09:49,222 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:09:49,247 - Server - INFO - Started up on port 50000
2018-03-09 17:09:49,266 - Server - INFO - Thread: 8944 - Start requesting peers. 2 peers currently
2018-03-09 17:09:49,285 - Server - WARNING - Thread: 9280 - Started updating from peers
2018-03-09 17:09:49,433 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:09:49,439 - Server - WARNING - Thread: 8944 - Complete requesting peers. 2 peers currently
2018-03-09 17:09:49,525 - Server - INFO - Thread: 9280 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:09:49,530 - Server - WARNING - Thread: 9280 - Completed updating from peers
2018-03-09 17:10:39,139 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:10:39,201 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:10:39,369 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:10:39,522 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:10:39,558 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:10:39,563 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:10:39,582 - Server - INFO - Started up on port 50000
2018-03-09 17:10:39,599 - Server - INFO - Thread: 4576 - Start requesting peers. 2 peers currently
2018-03-09 17:10:39,622 - Server - WARNING - Thread: 4912 - Started updating from peers
2018-03-09 17:10:39,802 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:10:39,808 - Server - WARNING - Thread: 4576 - Complete requesting peers. 2 peers currently
2018-03-09 17:10:39,899 - Server - INFO - Thread: 4912 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:10:39,905 - Server - WARNING - Thread: 4912 - Completed updating from peers
2018-03-09 17:10:42,086 - Server - ERROR - Thread: 4576 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49788 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 31, in print
    pubkey = serialization.load_pem_public_key(self.pub_key, backend=default_backend())
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/primitives/serialization.py", line 24, in load_pem_public_key
    return backend.load_pem_public_key(data)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1041, in load_pem_public_key
    self._handle_key_loading_error()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1292, in _handle_key_loading_error
    raise ValueError("Could not deserialize key data.")
ValueError: Could not deserialize key data.

2018-03-09 17:11:25,727 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:11:25,888 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:11:26,051 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:11:26,199 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:11:26,238 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:11:26,243 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:11:26,256 - Server - INFO - Started up on port 50000
2018-03-09 17:11:26,275 - Server - INFO - Thread: 5088 - Start requesting peers. 2 peers currently
2018-03-09 17:11:26,300 - Server - WARNING - Thread: 5424 - Started updating from peers
2018-03-09 17:11:26,457 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:11:26,464 - Server - WARNING - Thread: 5088 - Complete requesting peers. 2 peers currently
2018-03-09 17:11:26,554 - Server - INFO - Thread: 5424 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:11:26,560 - Server - WARNING - Thread: 5424 - Completed updating from peers
2018-03-09 17:11:42,646 - Server - ERROR - Thread: 5088 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:49836 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 75, in add_message_str
    msg.print()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 36, in print
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

2018-03-09 17:23:49,025 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:23:49,150 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:23:49,314 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:23:49,463 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:23:49,500 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:23:49,505 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:23:49,528 - Server - INFO - Started up on port 50000
2018-03-09 17:23:49,545 - Server - INFO - Thread: 9616 - Start requesting peers. 2 peers currently
2018-03-09 17:23:49,570 - Server - WARNING - Thread: 9952 - Started updating from peers
2018-03-09 17:23:49,706 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:23:49,712 - Server - WARNING - Thread: 9616 - Complete requesting peers. 2 peers currently
2018-03-09 17:23:49,816 - Server - INFO - Thread: 9952 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:23:49,822 - Server - WARNING - Thread: 9952 - Completed updating from peers
2018-03-09 17:23:59,416 - Server - ERROR - Thread: 9616 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50434 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 39, in verify
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

2018-03-09 17:26:58,660 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:26:58,773 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:26:58,941 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:26:59,081 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:26:59,115 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:26:59,120 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:26:59,144 - Server - INFO - Started up on port 50000
2018-03-09 17:26:59,162 - Server - INFO - Thread: 3312 - Start requesting peers. 2 peers currently
2018-03-09 17:26:59,186 - Server - WARNING - Thread: 3648 - Started updating from peers
2018-03-09 17:27:00,204 - Server - INFO - Thread: 3312 - Warning: Punishing peer 149.106.60.81.
2018-03-09 17:27:00,210 - Server - INFO - Thread: 3648 - Warning: Punishing peer 149.106.60.81.
2018-03-09 17:27:00,233 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 1)]
2018-03-09 17:27:00,270 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 2)]
2018-03-09 17:27:00,275 - Server - WARNING - Thread: 3648 - Completed updating from peers
2018-03-09 17:27:00,292 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 2)]
2018-03-09 17:27:00,299 - Server - WARNING - Thread: 3312 - Complete requesting peers. 2 peers currently
2018-03-09 17:27:04,284 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:27:04,385 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:27:04,539 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:27:04,691 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:27:04,732 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:27:04,737 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:27:04,748 - Server - ERROR - Unable to start on port 50000
2018-03-09 17:27:14,073 - Server - ERROR - Thread: 3312 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50628 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 33, in verify
    sigPlaintext = pubkey.decrypt(
AttributeError: '_RSAPublicKey' object has no attribute 'decrypt'

2018-03-09 17:28:43,507 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:28:43,618 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:28:43,774 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:28:43,918 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:28:43,953 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:28:43,953 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:28:43,972 - Server - INFO - Started up on port 50000
2018-03-09 17:28:43,990 - Server - INFO - Thread: 9232 - Start requesting peers. 2 peers currently
2018-03-09 17:28:44,010 - Server - WARNING - Thread: 9568 - Started updating from peers
2018-03-09 17:28:44,157 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:28:44,162 - Server - WARNING - Thread: 9232 - Complete requesting peers. 2 peers currently
2018-03-09 17:28:44,247 - Server - INFO - Thread: 9568 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:28:44,252 - Server - WARNING - Thread: 9568 - Completed updating from peers
2018-03-09 17:29:00,020 - Server - ERROR - Thread: 9232 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50698 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 40, in verify
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

2018-03-09 17:29:11,371 - Server - ERROR - Thread: 9232 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50710 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 40, in verify
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

2018-03-09 17:29:40,032 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:29:40,190 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:29:40,351 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:29:40,490 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:29:40,527 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:29:40,532 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:29:40,550 - Server - INFO - Started up on port 50000
2018-03-09 17:29:40,567 - Server - INFO - Thread: 928 - Start requesting peers. 2 peers currently
2018-03-09 17:29:40,593 - Server - WARNING - Thread: 1264 - Started updating from peers
2018-03-09 17:29:40,721 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:29:40,727 - Server - WARNING - Thread: 928 - Complete requesting peers. 2 peers currently
2018-03-09 17:29:40,847 - Server - INFO - Thread: 1264 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:29:40,853 - Server - WARNING - Thread: 1264 - Completed updating from peers
2018-03-09 17:29:44,050 - Server - ERROR - Thread: 928 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50734 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 40, in verify
    hashes.SHA512()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

2018-03-09 17:30:53,203 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:30:53,309 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:30:53,466 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:30:53,611 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:30:53,648 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:30:53,654 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:30:53,677 - Server - INFO - Started up on port 50000
2018-03-09 17:30:53,694 - Server - INFO - Thread: 464 - Start requesting peers. 2 peers currently
2018-03-09 17:30:53,715 - Server - WARNING - Thread: 800 - Started updating from peers
2018-03-09 17:30:53,858 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:30:53,864 - Server - WARNING - Thread: 464 - Complete requesting peers. 2 peers currently
2018-03-09 17:30:53,954 - Server - INFO - Thread: 800 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:30:53,960 - Server - WARNING - Thread: 800 - Completed updating from peers
2018-03-09 17:30:59,063 - Server - ERROR - Thread: 464 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50804 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 41, in verify
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 257, in _rsa_sig_verify
    backend._lib.EVP_PKEY_verify_init
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 191, in _rsa_sig_setup
    padding_enum = _rsa_sig_determine_padding(backend, key, padding, algorithm)
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 184, in _rsa_sig_determine_padding
    _Reasons.UNSUPPORTED_PADDING
cryptography.exceptions.UnsupportedAlgorithm: EME-OAEP is not supported by this backend.

2018-03-09 17:31:42,297 - Server - WARNING - =========== Server logging started ==========
2018-03-09 17:31:42,391 - Server - INFO - Reading peers from peers.txt.
2018-03-09 17:31:42,563 - Server - INFO - Thread: 9488 - Peer request accepted. andersm2-macpro.union.edu
2018-03-09 17:31:42,713 - Server - INFO - Thread: 9488 - Peer request accepted. 149.106.60.81
2018-03-09 17:31:42,752 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:31:42,758 - Server - WARNING - ============ Server init complete ===========
2018-03-09 17:31:42,769 - Server - INFO - Started up on port 50000
2018-03-09 17:31:42,787 - Server - INFO - Thread: 7792 - Start requesting peers. 2 peers currently
2018-03-09 17:31:42,810 - Server - WARNING - Thread: 8128 - Started updating from peers
2018-03-09 17:31:42,959 - Server - INFO - Wrote peers to peers.txt: [('149.106.60.81', 0)]
2018-03-09 17:31:42,965 - Server - WARNING - Thread: 7792 - Complete requesting peers. 2 peers currently
2018-03-09 17:31:43,047 - Server - INFO - Thread: 8128 - Received 11 blocks from 149.106.60.81.
2018-03-09 17:31:43,053 - Server - WARNING - Thread: 8128 - Completed updating from peers
2018-03-09 17:31:59,346 - Server - ERROR - Thread: 7792 - Command: MESSAGE_BROADCAST - From: 149.106.60.81:50870 - Exception in handling request
Traceback (most recent call last):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/network.py", line 422, in handle_connection
    elif self.blockchain.add_message_str(msg_str):
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/blockchain.py", line 76, in add_message_str
    msg.verify()
  File "/Users/Ben_Kopchains/Documents/Homework/CSC-483/Project_4/message.py", line 40, in verify
    hashes.SHA256()
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 475, in verify
    self._backend, padding, algorithm, self, signature, data
  File "/Users/Ben_Kopchains/Documents/Homework/__THESIS/Python Stats/venv/lib/python3.4/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 269, in _rsa_sig_verify
    raise InvalidSignature
cryptography.exceptions.InvalidSignature

